src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
```

Comment:
PullProofAI

There is a potential security vulnerability in the `FeedbackSecretGuard` class. Storing sensitive information like secrets directly in code can lead to exposure risks, especially if the code is version controlled or shared. It's recommended to avoid hardcoding secrets in code.

To address this issue, consider using environment variables or a secure secret management solution to store and retrieve sensitive information like the feedback secret dynamically. This approach enhances security by keeping secrets out of the codebase.

```diff
-     const secret = Env.SECRET.FEEDBACK;
+     const secret = process.env.FEEDBACK_SECRET; // Example of using environment variable
```

By implementing this change, you can improve the security of the application by separating sensitive data from the codebase and following best practices for handling secrets securely.


<!-- This is an auto-generated comment by PullProofAI -->